DZ_5_Тестирование_гипотез

In [2]:
import scipy.stats as stats
import numpy as np

#1. Когда используется критерий Стьюдента, а когда Z –критерий?
Ответ: \
данные критерии использутся для оценки нормального распредления \
**- если не известно std** (среднеквадр. отклонение ген/совокупности) **или D** (дисперсия ген совокупности) **и выборка менее или равна 30, то используется t-распределение и критерий Стьюдента**
**- в случае, когда std или D ген совокупности известна, а также в случае, когда std неизвестно, но выборка более 30, тогда** используется z-распределение и **находится z-критерий**

#2. Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

Решение: \
Запись данных:  X̅ = 17мм;  α = 0,05;  n = 100;  D = 4 кв.мм.; µ(x) = 17.5; std = √D = 2 \
**Запись гипотез: H0 = X̅ = µ(x) и H1 = X̅ > µ(x)** \
в случае, когда std неизвестно, но выборка более 30, тогда используется z-распределение \
SE = std / √n = 2/√100 = 0.2 \
z_raschetn = X̅ - µ(x)/ SE  z = 17.5 - 17 / 0.2 = 2,5 \

Ниже подгрузим z_critich:

In [12]:
 α = 0.05
 z_critich = stats.norm.ppf(1- α)
 print (round (z_critich,3))

1.645


Ответ: \
H0 остается если  z_raschetn < z_critich, а у нас z_raschetn = 2,5 > z_critich = 1.645 \
**таким образом выбираем H1:** \
**в выборке из n=100 шариков средний диаметр оказажется равным 17.5 мм в 95% случаев**


#3. Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

Решение: \
Запись данных:  X̅ = avg [202, 203, 199, 197, 195, 201, 200, 204, 194, 190] = 198.5  \
α = 1 - 0.99 = 0,01, но так как тест 2-сторонний, то α2 = 0,005;   df = n - 1 = 9;   n = 10   µ(x) = 200  D = ?  \
**Запись гипотез: H0 = µ(x) = 200 и H1 = X̅ ≠ µ(x)** \
в случае, когда std неизвестно, и выборка менее или равна 30, тогда используется t-распределение \
Дисперсию все-таки считал автоматически через np.std, σx= √D \
t_raschetn = (X̅ – μ) / (σx / √n) = (198.5 - 200) / (2.0555 / √10) =  −1.5 / (2.06 / 3.16) = −1.5 / 0,65 = -3 / 1,3 = - 2.3


Ниже подгрузим t_critich (табл/знач при  9 степ/свободы и ур стат/занчим. 1-α = 0.995):

In [6]:
X = [202, 203, 199, 197, 195, 201, 200, 204, 194, 190]
D = np.std(X)
σx = np.sqrt(D)
α, df = 0.005, 9
t_critich = stats.t.ppf(α, df)
print (f'D = {round(D, 4)}, σx = {round(σx, 4)}, t_critich = {round(t_critich, 3)}')

D = 4.2249, σx = 2.0555, t_critich = -3.25


Ответ:
H0 остается если t_raschetn < t_critich,  у нас t_raschetn = -2.3 < t_critich = -3,25 \
**таким образом выбираем H0:** \
**утверждение продавца, если учитывать, что доверительная вероятность равна 99% вес пачки равен 200гр - верно**

#4. Есть ли статистически значимые различия в росте дочерей? Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169, 165; Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160

In [7]:
M_height = [172, 177, 158, 170, 178, 175, 164, 160, 169, 165]
D_height = [173, 175, 162, 174, 175, 168, 155, 170, 160]
α = 0.05
print ('Гипотезы: \n H0 : Средние t-критерии до и после теста равны / H1 : Средние t-критерии до и после теста не равны / уровень pvalue = 0,05 ')
All_heights = M_height + D_height
array_key = {'Mothers_height': M_height, 'Daughters_height': D_height}
import random
if len(M_height) < len (D_height):
    random.seed(1)
    new_M = random.randint(min(M_height), max(M_height))
    M_height.append(new_M)
    print('Array:', next(key for key, value in array_key.items() if value == M_height), 'append with new value = ', new_M, '.  New length = ', len (M_height), 'elements')
elif len(M_height) > len (D_height):
    random.seed(1)
    new_D = random.randint(min(All_heights), max(All_heights))
    D_height.append(new_D)
    print('Array:', next(key for key, value in array_key.items() if value == D_height), 'append with new value = ', new_D, '.  New length = ', len (D_height), 'elements')
else:
    print ('All is OK, array length Mothers_height = Daughters_height')


X̅_M, X̅_D  = np.mean(M_height), np.mean(D_height)
print ('Avg height X̅_Mothers', X̅_M, '.  Avg height X̅_Daughters', X̅_D)
stats.ttest_rel(M_height, D_height) #выборки зависимые, поэтому ttest_rel

Гипотезы: 
 H0 : Средние t-критерии до и после теста равны / H1 : Средние t-критерии до и после теста не равны / уровень pvalue = 0,05 
Array: Daughters_height append with new value =  159 .  New length =  10 elements
Avg height X̅_Mothers 168.8 .  Avg height X̅_Daughters 167.1


TtestResult(statistic=0.8452006777465805, pvalue=0.4198999400452258, df=9)

In [13]:
pvalue = 0.4198999400452258
if pvalue < α:
    print ('Ответ: Поскольку pvalue меньше α, мы отвергаем H0 и принимаем H1')
else:
     print ('Ответ: Поскольку pvalue больше α, нет оснований опровергнуть H0')

Ответ: Поскольку pvalue больше α, нет оснований опровергнуть H0
